src/Development/IDE/Core/Compile.hs

```diff
@@ -247,10 +250,55 @@ upgradeWarningToError (nfp, sh, fd) =
   warn2err = T.intercalate ": error:" . T.splitOn ": warning:"
 
 hideDiag :: DynFlags -> (WarnReason, FileDiagnostic) -> (WarnReason, FileDiagnostic)
-hideDiag originalFlags (Reason warning, (nfp, _sh, fd))
-  | not (wopt warning originalFlags) = (Reason warning, (nfp, HideDiag, fd))
+hideDiag originalFlags (Reason warning, (nfp, sh, fd))
+  | not (wopt warning originalFlags)
+  = if warning `elem` unnecessaryDeprecationWarningFlags
+       then (Reason warning, (nfp, sh, fd{_severity = Just DsInfo}))
+       else (Reason warning, (nfp, HideDiag, fd))
```

Comment:
I assume HideDiag means they don't appear at all? Could you just look at _tags ahd only Hide the ones with no tags? (I've no idea here, somewhat guessing)

Reply:
I think that's a good idea: make those with tags "less important" but not hide them.

Implemented in the latest commit.